libraries/standard/mqtt/src/mqtt.c

```diff
@@ -504,10 +927,86 @@ MQTTStatus_t MQTT_Disconnect( MQTTContext_t * const pContext )
 
 /*-----------------------------------------------------------*/
 
-MQTTStatus_t MQTT_Process( MQTTContext_t * const pContext,
-                           uint32_t timeoutMs )
+MQTTStatus_t MQTT_ProcessLoop( MQTTContext_t * const pContext,
+                               uint32_t timeoutMs )
 {
-    return MQTTSuccess;
+    MQTTStatus_t status = MQTTBadParameter;
+    MQTTGetCurrentTimeFunc_t getCurrentTime = NULL;
+    uint32_t entryTime = 0;
+    MQTTPacketInfo_t incomingPacket;
+
+    if( pContext != NULL )
+    {
+        getCurrentTime = pContext->callbacks.getTime;
+        entryTime = getCurrentTime();
+        status = MQTTSuccess;
+    }
+    else
+    {
+        IotLogError( "MQTT Context cannot be NULL." );
+    }
+    
+    while( ( pContext != NULL ) && ( ( getCurrentTime() - entryTime ) < timeoutMs ) )
```

Comment:
getCurrentTime() can be implemented for MS resolution by using system start time as epoch ( or a choose an epoch so that we can fit in MS resolution). We don't need wallclock time. For now we will leave the usage like this since we will implement is when demo is being developed.

Reply:
Renamed to `getTimeStamp()` to avoid some possible confusion.